2008-12-29  Samuel Thibault  <samuel.thibault@ens-lyon.org>

	* ipc/ipc_thread.h (ipc_thread_enqueue_macro): Change IPC thread queue
	head to last added thread to favor locality.

diff -u -p -r1.1.1.1 ipc_thread.h
--- a/ipc/ipc_thread.h	25 Feb 1997 21:28:19 -0000	1.1.1.1
+++ b/ipc/ipc_thread.h	29 Dec 2008 20:45:57 -0000
@@ -103,6 +103,7 @@ MACRO_BEGIN								\
 		(thread)->ith_prev = _last;				\
 		_first->ith_prev = (thread);				\
 		_last->ith_next = (thread);				\
+		(queue)->ithq_base = (thread);				\
 	}								\
 MACRO_END
 
